### PR TITLE
UI: Replace themeID and style hints with class property

### PR DESCRIPTION
**Pull request by Warchamp7**
### Description

In the past we've used a custom property called themeID to apply special styling to widgets. These widgets could be targeted using attribute style CSS selectors like
`QWidget[themeID="addIcon"]`. This approach was flawed in instances where we wanted to apply multiple of these adjustments on a single widget.

To solve that, we adjusted to instead using a boolean value which could be targeted with `QWidget[addIcon="true"]`

However, we have recently discovered that QSS has (almost completely undocumented) special handling for when widgets have a custom property named `class`, allowing the class selector like CSS

```myWidget->setProperty("class", "addIcon toolButton")```

QSS
```
.addIcon {
    /* styles */
}

.toolButton {
    /* styles */
}
```

### Motivation and Context

Standardizes and unifies our approach for applying special appearance changes to widgets.

### How Has This Been Tested?

Ensured that the appearance of specific widgets all looks correct

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)
- Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [ ] I have included updates to all appropriate documentation.

### CODE REVIEW

File: shared/qt/wrappers/qt-wrappers.cpp [hunk 1]
```diff
@@ -318,10 +318,10 @@ void SetComboItemEnabled(QComboBox *c, int idx, bool enabled)
 			       : Qt::NoItemFlags);
 }
 
-void setThemeID(QWidget *widget, const QString &themeID)
+void setClasses(QWidget *widget, const QString &newClasses)
```
Comment (WizardCM) [inline shared/qt/wrappers/qt-wrappers.cpp:321]: Do we want to go with `setClasses` or something similar to CSS' `classList` which have `add` and `remove` utility functions with duplicate-detection? Feels a bit weird to me as a CSS user to completely purge the existing class without caring what the existing contents were.

Comment (Warchamp7) [reply shared/qt/wrappers/qt-wrappers.cpp:321]: I did something like that with the helper functions on #8447
https://github.com/obsproject/obs-studio/pull/8447/files#diff-1e0673933828dd8c9df583b80101ad9dfb838e9fefd8ad0aec4d28aae92d3bd5R91

I opted not to overhaul things on this particular PR but I think that is the road we should go down, yes

File: UI/forms/OBSAbout.ui [hunk 2]
```diff
@@ -87,6 +87,22 @@
          </property>
         </widget>
        </item>
+       <item>
+        <spacer name="verticalSpacer_2">
+         <property name="orientation">
+          <enum>Qt::Orientation::Vertical</enum>
+         </property>
+         <property name="sizeType">
+          <enum>QSizePolicy::Policy::Fixed</enum>
```
Comment (RytoEX) [inline UI/forms/OBSAbout.ui:96]: As far as I know, support for fully qualified enums was only backported to Qt 6.5/6.6. Ubuntu 24.04 still uses 6.4, so I'm not sure that it fully supports these, even though CI appears to be building. Ubuntu 24.10 and newer will use Qt 6.6+.

File: UI/data/themes/Yami.obt [hunk 3]
```diff
@@ -226,11 +233,151 @@
     *
     * Note: When using two operands that have a type (e.g. "px") the type must match!
     *       If only one operand has a type it'll be used for the result.
-    * Note 2: Cannot be !editable
+    * Note 2: Cannot be!editable
```
Comment (RytoEX) [inline UI/data/themes/Yami.obt:236]: This seems odd.

File: UI/data/themes/Yami.obt [hunk 4]
```diff
@@ -1193,13 +1296,15 @@ QSlider::handle {
 QSlider::handle:horizontal {
     height: 10px;
     width: 20px;
-    margin: -3px 0; /* handle is placed by default on the contents rect of the groove. Expand outside the groove */
+    /* handle is placed by default on the contents rect of the groove. Expand outside the groove */
```
Comment (RytoEX) [inline UI/data/themes/Yami.obt:1299]:
```suggestion
    /* Handle is placed by default on the contents rect of the groove. Expand outside the groove */
```

File: UI/data/themes/Yami.obt [hunk 5]
```diff
@@ -1193,13 +1296,15 @@ QSlider::handle {
...
+    margin: -3px 0;
 }
 
 QSlider::handle:vertical {
     width: 10px;
     height: 20px;
-    margin: 0 -3px; /* handle is placed by default on the contents rect of the groove. Expand outside the groove */
+    /* handle is placed by default on the contents rect of the groove. Expand outside the groove */
```
Comment (RytoEX) [inline UI/data/themes/Yami.obt:1306]:
```suggestion
    /* Handle is placed by default on the contents rect of the groove. Expand outside the groove */
```

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: UI/data/themes/Yami.obt (hunk 3 above)
Comment (Warchamp7) [reply UI/data/themes/Yami.obt:236]: My formatter botched that one. Will fix

**Review by RytoEX — APPROVED** [review]

**Pull request merged**